templates/docs/reference/bundled-checks.phtml

```diff
@@ -0,0 +1,258 @@
+            </p>
+        </li>
+        <li class="method-list-item">
+            <h3 class="service-list-item-label">PhpSchool\PhpWorkshop\Check\DatabaseCheck</h3>
+            <p class="service-list-item-description">
+                <dl class="check-description">
+                    <dt>Interface to implement:</dt>
+                    <dd><code>PhpSchool\PhpWorkshop\ExerciseCheck\DatabaseExerciseCheck</code></dd>
+                    <dt>Type:</dt>
+                    <dd><code>Listener</code></dd>
+                    <dt>Compatible Exercise Types:</dt>
+                    <dd><code>CLI, CGI</code></dd>
+                </dl>
+
+                This check sets up a database and a <code>PDO</code> object. It prepends the database DSN as a
+                CLI argument to the students submission so they can connect to the database. The PDO object is
+                passed to the exercise before and after the students submission has been executed, allowing you to
+                first seed the database and then verify the contents of the database.<br /><br />
+
+                <a class="link" href="#check-database">Learn how to use</a>
+            </p>
+        </li>
+    </ul>
+</div>
+
+<?= $this->renderContentHeader('check-functional-requirements', 'Functional Requirements Check') ?>
+<p>Here is an example of how to force the student to use the function <code>curl_exec</code> and ban the usage of
+<code>file_get_contents</code>. This could be useful if you wanted to teach advanced configuration of data transfers.</p>
+
+<pre><code class="lang-php">class MyExercise extends AbstractExercise implements ExerciseInterface, FunctionRequirementsExerciseCheck
+{
+    ...snip
+
+    /**
+     * @param ExerciseDispatcher $dispatcher
+     */
+    public function configure(ExerciseDispatcher $dispatcher)
+    {
+        $dispatcher->requireCheck(FunctionRequirementsCheck::class);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getRequiredFunctions()
+    {
+        return ['curl_exec']
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getBannedFunctions()
+    {
+        return ['file_get_contents']
+    }
+}
+</code></pre>
+
+<p>If a user submitted something like:</p>
+<pre><code class="lang-php">&lt;?php
+
+echo file_get_contents('http://example.com');
+</code></pre>
+
+<p>The output would be something along the lines of:</p>
+<a href="/img/func-requirements.png"><img src="/img/func-requirements.png" class="doc-terminal-screen"></a>
+
+<?= $this->renderContentHeader('composer-check', 'Composer Check') ?>
+<p>Here is an example of how to force the student to require the <code>nikic/fast-route</code> package via Composer. This
+is useful if you want to focus on a specific problem or promote popular/battle tested packages.</p>
+
+<pre><code class="lang-php">class MyExercise extends AbstractExercise implements ExerciseInterface, ComposerExerciseCheck
```

Comment:
@AydinHassan what do you think about putting the implements on newlines, they don't fit 100% in the sites container and it's not really explicit on all browsers that it's a scroll until you actually scroll 

http://ss.jhf.tw/Screen%20Shot%202016-05-16%20at%2021.38.32-H3FiMnbcIK.png